pkg/controller/component/lorry_utils.go

```diff
@@ -228,29 +202,46 @@ func buildLorryServiceContainer(synthesizeComp *SynthesizedComponent, container
 			})
 	}
 
-	container.Ports = []corev1.ContainerPort{
-		{
-			ContainerPort: int32(lorryHTTPPort),
-			Name:          constant.LorryHTTPPortName,
-			Protocol:      "TCP",
-		},
-		{
-			ContainerPort: int32(lorryGRPCPort),
-			Name:          constant.LorryGRPCPortName,
-			Protocol:      "TCP",
-		},
+	if len(synthesizeComp.PodSpec.Containers) > 0 {
+		mainContainer := synthesizeComp.PodSpec.Containers[0]
+		if len(mainContainer.Ports) > 0 {
+			port := mainContainer.Ports[0]
+			dbPort := port.ContainerPort
+			envs = append(envs, corev1.EnvVar{
+				Name:      constant.KBEnvServicePort,
+				Value:     strconv.Itoa(int(dbPort)),
+				ValueFrom: nil,
+			})
+		}
+
+		dataVolumeName := dataVolume
+		for _, v := range synthesizeComp.Volumes {
+			// TODO(xingran): how to convert needSnapshot to original volumeTypeData ?
```

Comment:
Can the user configured disk be directly mounted? Instead of distinguishing whether a snapshot is needed or not.

Reply:
yes, it is a TODO.